### DISCUSSION TITLE
Auto zoom with speed and distance

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#157] [Q&A] [answered]
There are two auto zoom modes: speed and distance.
if it were possible->
A mix of both would be ideal.
When approaching a roundabout or an intersection(where you have to turn), the zoom could change from speed
switch to distance.
That way you could get your bearings better.

**Comment by devemux86** [comment]
@Leeb64 thanks for the suggestion.

It seems an interesting combination, I remember it was discussed somewhere before.

- Distance auto zoom: It _zooms in_ more and more as we approach the next turn.
- Speed auto zoom: We usually slow down as we approach the next turn, so it should also zoom in.

Is this what you describe?

**Reply by Leeb64** [reply]
When approaching an out-of-town roundabout, this is shown as being quite small due to the higher speed.
If the zoom would zoom about 100-200m before reaching this roundabout so big that you can clearly see what's going on would be good.
The same for other changes of direction.
Especially out of town.

**Comment by Ciutnik** [comment]
This is a desirable idea. I found a discussion about this on the Kurviger Forum:

https://forum.kurviger.de/t/zoom-in-von-geschwindigkeit-und-entfernung-gleichzeitig-abhangig-machen/6395

**Comment by devemux86** [comment]
I have added it in my list for implementation. :slightly_smiling_face:

**Comment by devemux86** [comment]
"Speed + Distance" auto zoom was easy to implement:

<img src="https://github.com/devemux86/cruiser/assets/3484020/d7c22e98-5f4c-4d59-b2aa-a29494fc1e40" width="50%"/>

**Comment by devemux86** [comment]
At "Small distance" (300 m for car / motorcycle) before the next turn instruction, auto zoom changes from speed to distance.
Is this enough or do you prefer earlier in "Medium distance" (1.5 km for car / motorcycle)?

Remember that these distance can be changed in navigation settings.

<img src="https://github.com/devemux86/cruiser/assets/3484020/f82b5405-a759-42de-839d-a9b02ea41dcd" width="33%"/>

**Reply by Leeb64** [reply]
should work.

**Reply by Ciutnik** [reply]
I admit that it is difficult for me to imagine using three distances: Small, Medium and Large. Will it be possible to define three degrees of magnification (zoom) in these three distances?...

**Reply by devemux86** [reply]
The distance value for auto zoom only selects the point where the switch from speed to distance auto zoom happens.

Distance auto zoom works automatically: zoom more and more while keeping next turn instruction inside screen view.

**Reply by Ciutnik** [reply]
> The distance value for auto zoom only selects the point where the switch from speed to distance auto zoom happens.

Yes, this is fully understandable and expected. But why do we have three distances defined?

**Reply by devemux86** [reply]
> But why do we have three distances defined?

These settings allow to adjust the distances where navigation instructions are announced.
(different distances per navigation profile)

Please also see the related topic:
- https://github.com/devemux86/cruiser/discussions/108

Here we need to select a distance to switch between the 2 auto zoom modes.

**Reply by Ciutnik** [reply]
> These settings allow to adjust the distances of the navigation instructions.

Oh yes! Sorry, but I didn't realize because I was only focused on the auto-zoom function 😃 Maybe, so as not to confuse other users, there should be a value that is used only for this purpose, independent of the values ​​used for the instructions?

But maybe it's not necessary, so as not to multiply the settings, as long as there is some hint in which situations these distances are used.

Anyway, I can't wait to check it out in reality 👍

**Reply by devemux86** [reply]
> Anyway, I can't wait to check it out in reality 👍

**Cruiser 3.0.19** is already released, we wait Google Play to make it available everywhere.

Development always continues, this feature will be included in **Cruiser 3.0.20**. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
Why not also combining this with the speed reverse feature? It makes sense

**Comment by devemux86** [comment]
Auto zoom with _speed (normal / reverse) + distance_ will be available in the next Cruiser:

<img src="https://github.com/devemux86/cruiser/assets/3484020/ffe8cca5-e144-4ac2-b804-409f3784f49b" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
Auto zoom with speed + distance is available in [Cruiser 3.0.20](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Available in "Settings | Navigation | Auto zoom | Auto zoom mode" or search settings for "auto zoom".

**Reply by DolomitesTom** [reply]
already set the new option, but can't drive till the weekend 🥺.

**Comment by Cer-o** [comment]
I went out for a drive and took the opportunity to test the application.
Something that I did not like and that I would like to solve is the zoom when we approach a junction, roundabout or turn. Throughout the tour I have seen it too far away and when I reached the turn or roundabout, it did not zoom in, which makes it quite difficult to follow the instructions.

Some examples:
<img src="https://github.com/devemux86/cruiser/assets/122309655/ac38a234-fc14-4240-9532-84b74872b2cb" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122309655/59297680-1798-4767-b8b7-d0f3b4b81650" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/122309655/1c23d65e-7cda-474d-9f12-4dbc10e9746a" width="33%"/>

**Reply by devemux86** [reply]
@Cer-o thanks for the report.

What settings do you have in "Settings | Navigation | Auto zoom"?

Can you attach:
- A screenshot of your settings
- Or export app settings as .pref file

**Comment by Cer-o** [comment]
![image](https://github.com/devemux86/cruiser/assets/122309655/51d0b2f6-33bc-4cc4-b2c2-95051b62d561)
[configuración.pref.zip](https://github.com/devemux86/cruiser/files/11635777/configuracion.pref.zip)

**Reply by devemux86** [reply]
@Cer-o thanks for the extra details.

(you have "Distance" in auto zoom mode, default is "Speed")

With "Distance" selection, navigation zooms in as we approach the next turn instruction (yellow markers).

You can select "Speed" in auto zoom mode, if you want the auto zoom to depend on your speed:
- Low speed: zoom in
- High speed: zoom out

Note that the min / max selections limit the auto zoom, so you could change these settings too.

**Comment by Cer-o** [comment]
So you advise me to choose "Speed"?
Is the minimum and maximum zoom correct?
Have you taken a look at the rest of the configuration in case I could improve it in any way?

**Reply by devemux86** [reply]
Better start with the default "Speed" auto zoom and see how it goes with your driving.
If you want navigation to auto zoom less or more, you can change its min / max limits.

It is recommended to use the default Cruiser settings and make changes as you learn the app.
You can always export your settings for backup, reset them to make tests and restore them later.
